modules/nlu/src/backend/api.ts

```diff
@@ -205,6 +219,27 @@ export default async (bp: typeof sdk, state: NLUState) => {
     const { botId, id } = req.params
     const botEngine = state.nluByBot[botId].engine1 as ScopedEngine
     await botEngine.storage.deleteEntity(id)
+
+    const intents = await botEngine.storage.getIntents()
+    const awaitsList = intents.map(async intent => {
+      const entitySlotIndex = intent.slots.findIndex(slot => slot.entities.includes(id))
```

Comment:
This will only find the first slot having the corresponding entity. For example:
Given a **find flight** intent with `airport_from` and `airport_to` slots assigned to and `airport_code` entity.

If the `airport_code` entity is deleted, we want to remove it from both `airport_from` and `airport_to` slots not the first found.

Does that makes sens ?

Reply:
Yep, ty, fixed